tests/types/test_arguments.py

```diff
@@ -200,3 +205,125 @@ class RootQuery(NameQuery, ExampleQuery):
     assert definition.fields[1].arguments[0].name == "id"
     assert definition.fields[1].arguments[0].type == strawberry.ID
     assert definition.fields[1].arguments[0].is_optional is False
+
+
+def test_annotated_argument_on_resolver():
+    @strawberry.type
+    class Query:
+        @strawberry.field
+        def name(  # type: ignore
+            argument: Annotated[
+                str,
+                strawberry.argument(description="This is a description"),  # noqa: F722
+            ]
+        ) -> str:
+            return "Name"
+
+    definition = Query._type_definition
+
+    assert definition.name == "Query"
+
+    assert len(definition.fields[0].arguments) == 1
+
+    argument = definition.fields[0].arguments[0]
+
+    assert argument.name == "argument"
+    assert argument.type == str
+    assert argument.is_optional is False
+    assert argument.description == "This is a description"
+
+
+def test_annotated_optional_arguments_on_resolver():
+    @strawberry.type
+    class Query:
+        @strawberry.field
+        def name(  # type: ignore
+            argument: Annotated[
+                Optional[str],
+                strawberry.argument(description="This is a description"),  # noqa: F722
+            ]
+        ) -> str:
+            return "Name"
+
+    definition = Query._type_definition
+
+    assert definition.name == "Query"
+
+    assert len(definition.fields[0].arguments) == 1
+
+    argument = definition.fields[0].arguments[0]
+
+    assert argument.name == "argument"
+    assert argument.type == str
+    assert argument.is_optional is True
+    assert argument.description == "This is a description"
+
+
+def test_annotated_argument_with_default_value():
+    @strawberry.type
+    class Query:
+        @strawberry.field
+        def name(  # type: ignore
+            argument: Annotated[
```

Comment:
should this have self here? 😊 (same for the other methods)

Reply:
Well it could but it's not necessary. TBH I find it quite misleading to have `self` on resolvers because it's not necessary an instance of the type and calling it `self` implies that it always will be. For example on the `Query` type `self` will always be `None` unless you pass a root value to `execute`:

```python
@strawberry.type
class Query:
	@strawberry.field
	def hello(self, name: str) -> str:
		assert self is None
		return f"Hi {name}"

schema = strawberry.Schema(Query)
schema.execute_sync(""" { hello } """)
```

Reply:
that's true, but I wonder if we should make it required, I feel like making self optional is quite confusing 😊